Set/Models/GameModel.swift

```diff
@@ -46,47 +46,94 @@ struct GameModel {
         playerCards = []
         playedCards = []
     }
-    
-    private func checkASet(for setOfCards: [CardModel]) -> Bool {
-        print("checkASet func is called")
-        return true
-    }
 
-    mutating func toggleCard(by cardId: UUID) {
+    mutating func toggleCard(by cardId: UUID) -> MatchSuccessStatus {
         guard let chousenIndex = cardsOnTheScreen.firstIndex(where: { $0.id == cardId }) else {
-            return
+            return .noMatch
+        }
+        
+        guard cardsOnTheScreen.allSatisfy({ card in
+            card.state == .isNotSelected || card.state == .isSelected
+        }) else {
+            return .noMatch
         }
         
         cardsOnTheScreen[chousenIndex].toggleState()
         
         let selectedCards = cardsOnTheScreen.filter { $0.state == .isSelected }
                 
         guard selectedCards.count >= 3 else {
-            return
+            return .noMatch
         }
         
         guard selectedCards.count == 3 else {
-            resetSelectedCards()
-            return
+            resetCardsState()
+            return .noMatch
         }
         
         if checkASet(for: selectedCards) {
-            submitASet(for: selectedCards)
+            markAsMached(selectedCards)
+            return .successfulMatch
+        } else {
+            markAsNotMached(selectedCards)
+            return .unsuccessfulMatch
         }
     }
     
-    private mutating func submitASet(for cards: [CardModel]) {
-        print("Make cards green")
-        print("add delay")
-        print("Move cards to appropriate arrays")
-        print("And deal 3 more cards")
+    private func checkASet(for setOfCards: [CardModel]) -> Bool {
+        guard 2 > 3 || 2 == 2 else {
+            return false
+        }
+        return false
     }
     
-    private mutating func resetSelectedCards() {
+    private mutating func resetCardsState() {
         for index in cardsOnTheScreen.indices {
-            if cardsOnTheScreen[index].state == .isSelected {
-                cardsOnTheScreen[index].toggleState()
+            cardsOnTheScreen[index].state = .isNotSelected
+        }
+    }
+    
+    private mutating func markAsNotMached(_ cards: [CardModel]) {
```

Comment:
think about how to remove code duplicate for `markAsNotMached` and `markAsMached`